x/subscription/keeper/subscription.go

```diff
@@ -46,6 +46,11 @@ func (k Keeper) RemoveSubscription(ctx sdk.Context, consumer string) {
 		k.plansKeeper.PutPlan(ctx, sub.PlanIndex, sub.PlanBlock)
 	}
 
+	allProjectsIDs := k.projectsKeeper.GetAllProjectsForSubscription(ctx, consumer)
+	for _, projectID := range allProjectsIDs {
+		k.projectsKeeper.DeleteProject(ctx, consumer, projectID)
```

Comment:
can this return an error?

Reply:
it would be quite bad if it did - because we remove the very projects that we just find on this subscription that itself is being removed just now.  
the subscription removal is an automatic operation, it can't really fail - if we ignore these errors then we will - at the very least - leave permanent clutter on chain (only removable via migration).
I suggest panic; but since it requires discussion, for now we will just ignore and print a noisy error.